core/src/test/java/org/elasticsearch/test/TestSearchContext.java

```diff
@@ -284,8 +285,8 @@ public ContextIndexSearcher searcher() {
         return searcher;
     }
 
-    public void setSearcher(ContextIndexSearcher searcher) {
-        this.searcher = searcher;
+    public void setSearcher(Engine.Searcher searcher) {
+        this.searcher = new ContextIndexSearcher(this, searcher);
```

Comment:
This is a bit cleaner for the 1.x parent/child queries lucene level tests, so I didn't need to repeat the creation of the ContextIndexSearcher 5 times.